.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,55 @@
+name: unit-tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  run-tests:
+    runs-on: [ self-hosted ]
```

Comment:
If you do have test problems, this script will help you find the fewest tests needed to replicate: https://github.com/edx/edx-platform/blob/master/scripts/xdist/find_order_dependent_test_failures.py
However, it needs to be updated to get a different input that is not xdist. If you get a chance to fix the script, it may help you for this effort and others in the future.

Reply:
this would require some discovery work to optionally run the checks with GitHub runners on forks, the way I see it forks would have to enable these checks and make the change from `self-hosted` to `Ubuntu` ?